packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -0,0 +1,72 @@
+import {Component, Prop, h, Listen, State} from '@stencil/core';
+import {Result} from '@coveo/headless';
+import {LinkWithResultAnalytics} from '../result-link/result-link';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {ResultContextEvent} from '../result-template-components/result-template-decorators';
+
+/**
+ * @part source-url
+ * @part source-title
+ * @internal
+ */
+@Component({
+  tag: 'atomic-smart-snippet-source',
+  shadow: false,
+})
+export class AtomicSmartSnippetSource implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Prop({reflect: true, mutable: true}) source!: Result;
+
+  @Prop() selectSource!: () => void;
+  @Prop() beginDelayedSelectSource!: () => void;
+  @Prop() cancelPendingSelectSource!: () => void;
```

Comment:
Do you mean for those to be events or is this intentional?

```suggestion
  @Event() selectSource!: EventEmitter;
  @Event() beginDelayedSelectSource!: EventEmitter;
  @Event() cancelPendingSelectSource!: EventEmitter;
```

Reply:
Since it's an internal component & it's a direct child, using props is fine I think, like if it was a functional component